corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/split_screen_case_search_spec.js

```diff
@@ -0,0 +1,119 @@
+/* eslint-env mocha sinon */
+/* global Backbone */
+
+describe('Split Screen Case Search', function () {
+    const API = hqImport("cloudcare/js/formplayer/menus/api"),
+        Controller = hqImport('cloudcare/js/formplayer/menus/controller'),
+        FakeFormplayer = hqImport("cloudcare/js/formplayer/spec/fake_formplayer"),
+        FormplayerFrontend = hqImport('cloudcare/js/formplayer/app'),
+        Toggles = hqImport('hqwebapp/js/toggles'),
+        Utils = hqImport('cloudcare/js/formplayer/utils/utils'),
+        stubs = {};
+
+    const REGIONS = {
+        main: 'main',
+        sidebar: 'sidebar',
+    };
+
+    let splitScreenCaseListView,
+        sandbox,
+        currentUrl,
+        getRegion;
+
+    beforeEach(function () {
+        splitScreenCaseListView = hqImport('cloudcare/js/formplayer/spec/fixtures/split_screen_case_list');
```

Comment:
Adding this in `beforeEach` isn't doing what I hoped it would do. Is there a nice way to get a clean version of this fixture each time rather than retaining modifications from one test to the next? Looking at my `splitScreenCaseListView.type = '';` and `splitScreenCaseListView.type = 'query';` and realizing those changes are being kept between tests.

Reply:
huh it's been a long time since I've done js tests, but this is what I would expect to work. do you know if it's caching between tests?

Reply:
In non-requirejs areas of HQ like web apps, js is included as a script tag, and when the browser hits the script tag, it runs the file's code, which means calling `hqDefine`, which executes the module's code and adds its return value to a big global object `COMMCAREHQ_MODULES`. This global tracks all js modules, using module names as keys and module return values as values (see [here](https://github.com/dimagi/commcare-hq/blob/aeb9b8f51f481436bdc97fd9bacd18bcbf7c8054/corehq/apps/hqwebapp/static/hqwebapp/js/hqModules.js#L85)). Calls to `hqImport` then just access `COMMCAREHQ_MODULES` (see [here](https://github.com/dimagi/commcare-hq/blob/aeb9b8f51f481436bdc97fd9bacd18bcbf7c8054/corehq/apps/hqwebapp/static/hqwebapp/js/hqModules.js#L103)). For `split_screen_case_list.js`, the return value is the big fake response object that `makeEntitiesResponse` creates. So that object only gets created once, and any changes you make to it will indeed affect all future code.

Two possible resolutions come to mind:
1. Treat the fake response like the global it is: any changes you make in a test, undo them at the end of the test.
2. Update `split_screen_case_list.js` so that instead of returning an object, you wrap the existing code in a creation function, and then call that function within tests to get a fresh version of the response. So something like:
```
hqDefine("cloudcare/js/formplayer/spec/fixtures/split_screen_case_list", function () {
    const FakeFormplayer = hqImport("cloudcare/js/formplayer/spec/fake_formplayer");

    const createResponse = function () {
        return FakeFormplayer.makeEntitiesResponse({
            "title": "Search All Cases",
            ...
        });
    };

    return {
        createResponse: createResponse,
    };
});
```

Reply:
@orangejenny That makes a lot of sense, thank you for the explanation! I think I'm opting for something closer to the first solution. It seems like I should be able to use a `_.omit()` for most of these, to pass in a modified version of the object without touching the original.